sunpy/io/tests/test_fits.py

```diff
@@ -176,3 +176,10 @@ def test_warn_longkey():
                                'goodkey': 'test'})
     assert 'GOODKEY' in fits.keys()
     assert 'BADLONGKEY' not in fits.keys()
+
+
+def test_read_memmap():
+    # Test that memmap is passed correctly to the FITS reader
+    data, header = sunpy.io._fits.read(AIA_171_IMAGE, memmap=True)[0]
+    assert isinstance(data, np.ndarray)
```

Comment:
So this isnt a memory mapped array?

Reply:
I thought, data is a ndarray together with io.header it makes memmap.

Reply:
data, header pair is what the reader returns. I had assumed that a correctly memory mapped array would have a different type?

Reply:
I think, we can use [np.all](https://numpy.org/doc/stable/reference/generated/numpy.all.html) to check whether both gives same result, 
So, the test will be like:
 ```py
    # Test that memmap=False gives the same data as memmap=True
    data, header = sunpy.io.read_file(AIA_171_IMAGE, memmap=False)[0]
    data_memmap, header_memmap = sunpy.io.read_file(AIA_171_IMAGE, memmap=True)[0]
    assert np.all(data == data_memmap)
    assert np.all(header == header_memmap)
```

Reply:
Testing for value equality is not a useful test.  The way that memory mapping works in `astropy.io.fits` is that a `mmap.mmap` instance is created and then passed as a buffer to `np.ndarray()`.  What you need to do is to find and check the base buffer.  Here's a simple utility to check whether a `mmap.mmap` instance is being used:
```python
def is_memmap(data):
    if data.base is None:
        return False
    return isinstance(data.base, mmap.mmap) or is_memmap(data.base)
```
Be aware that reading in a FITS file that uses image compression or image scaling will – by default – defeat memory mapping and create the whole data array in memory.  For example, you need to open `AIA_171_IMAGE` with `disable_image_compression=True` to actually use memory mapping.  (To disable image scaling, you'd pass in `do_not_scale_image_data=True`.)

As an aside, you shouldn't use sample data like `AIA_171_IMAGE` for unit tests.  We have specific test files.

Reply:
Just to note (as I made the same assumption), the `AIA_171_IMAGE` is actually a test file.

Reply:
Oh, sorry, okay.  The variable shouldn't be named that to avoid confusion for code maintenance.

Reply:
It does need to be renamed.

Reply:
> To disable image compression, you'd pass in `disable_image_compression=True`. To disable image scaling, you'd pass in `do_not_scale_image_data=True`

Pass them to the `sunpy.io._fits.read()` or `sunpy.io.read_file()` function? (I don't find anything related to these arguments in the sunpy documentation)

Reply:
\<sigh\>  Apparently neither.  For some reason, `sunpy.io._fits.read()` (which is called by `sunpy.io.read_file()`) fails to pass `kwargs` on to the `astropy.io.fits.open()` call.  

Fortunately, you don't need to disable either for the AIA test file, so it's not necessary for this PR.

Reply:
I have a PR for this issue: https://github.com/sunpy/sunpy/pull/6123

Reply:
> def is_memmap(data):

Can I add this function somewhere and then import it into the test files?

Reply:
Yes that is ok.

Reply:
> It does need to be renamed and we can do it in this PR.


- `RHESSI_IMAGE` to `TEST_RHESSI_IMAGE`
- `AIA_171_IMAGE` to `TEST_AIA_IMAGE`
- `EIT_195_IMAGE` to `TEST_EIT_HEADER`
- `SWAP_LEVEL1_IMAGE` to `TEST_SWAP_HEADER` 

Should I do these renames?

Reply:
Sure